src/shared_modules/utils/registryHelper.h

```diff
@@ -108,6 +108,37 @@ namespace Utils
                 return ret;
             }
 
+            std::vector<std::string> enumerateValueKey() const
+            {
+                std::vector<std::string> ret;
+                constexpr auto MAX_KEY_NAME_VALUE {260}; // https://docs.microsoft.com/en-us/windows/win32/sysinfo/registry-element-size-limits
```

Comment:
![image](https://user-images.githubusercontent.com/14300208/154601116-1e35e1ba-b6dc-4d4d-b60a-fb064e198527.png)
https://docs.microsoft.com/en-us/windows/win32/sysinfo/enumerating-registry-subkeys

fully diffuse the documentation, let's expand this buffer to 32kb.

Reply:
One possible option is to call the function twice, once with an insufficient buffer to get the correct size that is required to be set aside.



![image](https://user-images.githubusercontent.com/14300208/154601659-0e607557-07d0-4456-914f-920a540ce059.png)

Reply:
Hi @Dwordcito 

The second option of using the function twice to get the key size was discarded because `RegEnumValue` never delivers the correct size when the buffer is insufficient.

Best regards
Hanes